src/UI/Component/Modal/Modal.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace ILIAS\UI\Component\Modal;
+
+use ILIAS\UI\Component\Component;
+use ILIAS\UI\Component\JavaScriptBindable;
+use ILIAS\UI\Component\Trigger\Triggerable;
+use ILIAS\UI\Component\Trigger\TriggerAction;
+
+/**
+ * This describes commonalities between the different modals
+ */
+interface Modal extends Component, JavaScriptBindable, Triggerable
+{
+
+    /**
+     * Get the title of the modal
+     *
+     * @return string
+     */
+    public function getTitle();
+
+
+    /**
+     * Get the component representing the content of the modal
+     *
+     * @return \ILIAS\UI\Component\Component
+     */
+    public function getContent();
+
+
+    /**
+     * Get all buttons of the modal
+     *
+     * @return \ILIAS\UI\Component\Button\Button[]
+     */
+    public function getButtons();
+
+
+    /**
+     * Get a modal like this with another title
+     *
+     * @param string $title
+     * @return Modal
+     */
+    public function withTitle($title);
+
+
+    /**
+     * Get a modal like this with another content
+     *
+     * @param \ILIAS\UI\Component\Component $content
+     * @return Modal
+     */
+    public function withContent(Component $content);
+
+
+    /**
+     * @return TriggerAction
+     */
+    public function show();
```

Comment:
The `show()` method returns the action to open the modal in the frontend.
```
$modal = $modal->withTitle("Foo");
$button->triggerAction($modal->show());
$modal->withTitle("Bar");
```
Since `$modal->withTitle("Bar");` returns a new instance of a modal, the button would show the modal with title "Foo".